### PR TITLE
Crm 19534 - Avoid db constraint violation when saving report

### PR DESCRIPTION
**Pull request by jmcclelland**
Only happens to non-admins

### CODE REVIEW

File: CRM/Core/BAO/Dashboard.php [hunk 1]
```diff
@@ -431,31 +431,37 @@ public static function addContactDashlet($dashlet) {
...
       $dao = CRM_Core_DAO::executeQuery($query);
       while ($dao->fetch()) {
-        $contactIDs[] = $dao->contact_id;
+        $contactIDs[$dao->contact_id] = NULL;
       }
     }
     else {
       //Get the id of Logged in User
       $session = CRM_Core_Session::singleton();
+      $contactID = CRM_Core_Session::singleton()->getLoggedInContactID();
```
Comment (seamuslee001) [inline CRM/Core/BAO/Dashboard.php]: @jmcclelland Jamie shouldn't the line below be removed as well?

Comment (colemanw) [inline CRM/Core/BAO/Dashboard.php]: This is a static function, shouldn't be called non-statically.

**Comment by jmcclelland** [comment]
Thanks @colemanw and @seamuslee001 for the feedback (and sorry for the sloppy commit). It should be cleaned up now.

**Comment by monishdeb** [comment]
@jmcclelland there is style warning highlighted in the test build https://test.civicrm.org/job/CiviCRM-Core-PR/12350/checkstyleResult/new/ 

``` Dashboard.php:455, ControlSignature, Priority: High
Expected "if (...) {\n"; found "if(...) {\n" ```

**Comment by jitendrapurohit** [comment]
Cherry-picked all the commits and created a new PR https://github.com/civicrm/civicrm-core/pull/9456 including jenkin warning fix and unit test to check the error.

**Comment by monishdeb** [comment]
Closed in favor of https://github.com/civicrm/civicrm-core/pull/9456

**Pull request closed without merging**